javascript/javascript3/week1/weather.js

```diff
@@ -0,0 +1,126 @@
+const body = document.querySelector("body");
+const input = document.getElementById("input-city");
+const button = document.getElementById("button");
+
+button.addEventListener("click", getCityWeatherInfo);
+const apiKey = "aa5a826cc684a7127fa32b9ff1a5cf8e";
+
+//get data from API by city name
+function getCityWeatherInfo() {
+  let cityName = input.value;
+
+  if (!cityName) {
+    alert("please insert a city");
+  }
+
+  fetch(
+    `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=${apiKey}`
+  )
+    .then((response) => response.json())
+    .then((data) => {
+      if (data.cod === "404") {
+        alert("It is not a city name");
+      }
+      renderWeatherInfo(data);
+    });
+}
+
+//all callbacks,render weather info
+function renderWeatherInfo(data) {
+  renderCityName(data.name);
+  renderCityTempreture(Math.round(data.main.feels_like - 273.15));
+  renderWeatherType(data.weather[0].icon);
+  renderWindSpeed(data.wind.speed);
+  renderHowCloudy(data.clouds.all, data.weather[0].description);
+  renderSunrise(new Date(data.sys.sunrise * 1000));
+  renderSunset(new Date(data.sys.sunset * 1000));
+}
+
+//all functions
+const h2 = document.createElement("h2");
+function renderCityName(nameOfCity) {
+  h2.innerHTML = nameOfCity;
+  body.appendChild(h2);
+}
+
+const fH6 = document.createElement("h6");
+function renderCityTempreture(tempreture) {
+  fH6.innerHTML = `tempreture feels like: ${tempreture} &deg c`;
+  h2.appendChild(fH6);
+}
+const image = document.createElement("img");
+function renderWeatherType(icon) {
+  image.src = ` http://openweathermap.org/img/wn/${icon}@2x.png`;
+  body.appendChild(image);
+}
+const sH6 = document.createElement("h6");
```

Comment:
Your variable names most be more descriptive. Maybe `tempretureHeader` instead of `fH6`